src/main/java/com/example/elasticsearch/SearchService.java

```diff
@@ -20,25 +19,22 @@
 public class SearchService {
 
     private WebClient client;
-    private final StoreService service;
 
 
-    public SearchService(Vertx vertx, ElasticAddress address, StoreService service) {
-        this.service = service;
+    public SearchService(Vertx vertx, ElasticAddress address) {
         this.client = WebClient.create(vertx, new WebClientOptions()
                 .setDefaultPort(address.getPort())
                 .setDefaultHost(address.getHost()));
     }
 
     public Uni<JsonObject> searchItemsBySearchRequest(SearchRequest request) {
-
         SearchSourceBuilder ssb = new SearchSourceBuilder();
 
         String query = ssb.query(QueryBuilders.boolQuery()
-                .must(request.getCategoryId() == null || request.getCategoryId() == 0 ?
+                .must(request.getCategoryId() == null || request.getCategoryId() == 0L ?

```

Comment:
How about not to duplicate this check, and have it in one place. It can be extracted to some util class, and used globally.
```suggestion
                    private boolean isEmpty(Long categoryId) {
        return categoryId == null || categoryId == 0L;
    }

```

We could go beyond that method, and it could be turned to `ifElse`, getting `isEmpty` as a first parameter, and two predicates: `matchAllQuery` and `matchQuery`, returning either depending of outcome of first method.

Reply:
This could be refactored to:
```
                .must(ifTrueOrElse(isEmpty(request.getCategoryId()),
                        () -> matchAllQuery(),
                        () -> matchQuery("categoryIds", request.getCategoryId()))
                )
```
where:
```
    private boolean isEmpty(Long categoryId) {
        return categoryId == null || categoryId == 0L;
    }
```

and:

```
private QueryBuilder ifTrueOrElse(Boolean matchAll,
                                      Supplier<QueryBuilder> all,
                                      Supplier<QueryBuilder> other) {
        return matchAll ? all.get() : other.get();
    }
```

Which is more readable to you?

Reply:
Second idea is much more readable, but not as obvious to write, great one! Implemented!